src/main/scala/zio/prelude/Equal.scala

```diff
@@ -838,11 +838,17 @@ object Equal extends Lawful[Equal] {
 
   /**
    * Equality for `Throwable` values.
-   * Comparison is based on: Class, message and cause (stack trace is ignored).
+   * Comparison is based on: Class, message, cause and stack trace.
    */
   implicit val ThrowableEqual: Equal[Throwable] =
```

Comment:
We had a previous discussion in #211 about not requiring the same stack trace here. Equality for `Throwable` values is not very well defined but not obvious to me that this is an improvement.

Reply:
Right. I think the other side of that is that it can make it very difficult to state any properties regarding values that contain `Throwable`. For example we might like to say something like:

```scala
Future(throw new Exception("boom")).value === Some(Failure(new Exception("boom")))
```

But this isn't necessarily true because these exceptions may contain different stack traces. Essentially the `Throwable` carries along this metadata about the history of the computation in addition to just the value itself. The stack trace is definitely useful for debugging purposes but I think it is questionable whether that should be part of the definition of equality or be regarded as more an implementation detail like the order in which parallel failures occurred (which again could be very useful for debugging but would not consider in the definition of equality).

It is probably less of a problem for ZIO itself because we are polymorphic in our error type but I think for anyone who uses `Throwable` as part of their data types it would be pretty painful.

Reply:
I've removed the check of stack traces